tests/decomposition/test_decomposition_graph.py

```diff
@@ -459,3 +479,28 @@ def custom_decomp(phi, wires):
         assert graph.resource_estimate(op) == to_resources(
             {qml.H: 1, qml.CNOT: 2, qml.RX: 1, qml.PhaseShift: 1}
         )
+
+    def test_pow_pow(self, _):
+        """Tests nested power decompositions."""
+
+        op = qml.pow(qml.pow(qml.H(0), 3), 2)
+        graph = DecompositionGraph(operations=[op], target_gate_set={"Hadamard"})
+        # 3 operator nodes: Pow(Pow(H)), Pow(H), and H
+        # 2 decomposition nodes for Pow(Pow(H)) and Pow(H)
+        assert len(graph._graph.nodes()) == 5
+        # 2 edges from decompositions to ops and 2 edges from ops to decompositions
+        assert len(graph._graph.edges()) == 4
+
+        graph.solve()
+        with qml.queuing.AnnotatedQueue() as q:
+            graph.decomposition(op)(*op.parameters, wires=op.wires, **op.hyperparameters)
+
+        assert q.queue == [qml.pow(qml.H(0), 6)]
```

Comment:
Just to confirm: The graph decomposition algorithm should _not_ allow to decompose `qml.pow(qml.H(0), z)` for some `z!=6` now, should it? :)

Reply:
Correct, it shouldn't